File: Assets/Selectable.js
var selectedMaterial : Material;
var defaultMaterial : Material;
var selected : boolean = false;

defaultMaterial = renderer.material;

function OnMouseDown (){
	if (selected) {
		unSelect();
	} else {
		select();
	}
}

function select() {
	
		for(var endzone : GameObject in GameObject.FindGameObjectsWithTag("Endzone"))
{
    endzone.GetComponent(Selectable).unSelect();
}
GameObject.Find("Game").GetComponent(GameLogic).target = this.transform;
		renderer.material = selectedMaterial;
		selected = true;
}

function unSelect() {
	renderer.material = defaultMaterial;
		selected = false;
}

File: Assets/GameAssets/Scripts/UI/Selectable.cs
using UnityEngine;
using System.Collections;

public class Selectable : MonoBehaviour {
	private Renderer renderer;
	
	void Start() {
	}
	
	public void Select(Material m) {
		transform.Find("Selector").GetComponent<MeshRenderer>().material = m;
		transform.Find("Selector").GetComponent<MeshRenderer>().enabled = true;
		// transform.GetComponent<Route>().ShowMesh();
	}
	
	public void UnSelect() {
		transform.Find("Selector").GetComponent<MeshRenderer>().enabled = false;
		// transform.GetComponent<Route>().HideMesh();
	}
}
